src/core/include/units/quantity_cast.h

```diff
@@ -87,6 +87,50 @@ struct cast_traits<From, To> {
   using rep_type = std::common_type_t<From, To>;
 };
 
+template<Quantity To, typename D, typename U, typename Rep>
+constexpr bool is_cast_noexcept() noexcept
+{
+  using traits = detail::cast_traits<Rep, typename To::rep>;
+  using ratio_type = TYPENAME traits::ratio_type;
+  using rep_type = TYPENAME traits::rep_type;
+  using calc_type = decltype(std::declval<rep_type>() * std::declval<ratio_type>() / std::declval<ratio_type>() *
+                             std::declval<ratio_type>());
+
+  return std::is_nothrow_constructible_v<rep_type, Rep>&& noexcept(
```

Comment:
Here we have a bug in clang-format, which I'm going to fix very soonish. We can wait for the fix to merge this, or reformat afterwards. No idea how you guys handle such scenarios.

Reply:
Again, why do we need such a complicated `noexcept` calculation here? It breaks the LEWG `noexcept` policy and thus will probably not be accepted for standardization anyway.

Reply:
I don't know about the LEWG `noexcept` policy.
I just wanted to make sure that it's only marked as `noexcept`, if it really is.

I'm happy to add more `noexcept` on functions like `common`, `relative`, etc. and add a concept to check if the arithmetic on a type is `noexcept`. That would allow to use a fairly simple `noexcept` specification on `quantity_cast`.

Reply:
For the remaining warnings, we might consider doing something similar to this: https://github.com/mpark/variant/commit/7d9454be276ad69e407db6e51aab98c995a8cc00 but with our own macros that you already know.

Reply:
Nice read.
But from that document the default CTor for example shouldn't be marked as `noexcept`.

Reply:
It can, assuming it is an unconditional `noexcept`. Actually, `= default` will "magically" put `noexcept` if only possible.